experimental/traffic-portal/src/app/core/cache-groups/regions/table/regions-table.component.ts

```diff
@@ -0,0 +1,139 @@
+* limitations under the License.
+*/
+
+import { Component, OnInit } from "@angular/core";
+import { FormControl } from "@angular/forms";
+import { MatDialog } from "@angular/material/dialog";
+import { ActivatedRoute, Router } from "@angular/router";
+import { BehaviorSubject } from "rxjs";
+import { Region, ResponseRegion } from "trafficops-types";
+
+import { CacheGroupService } from "src/app/api";
+import { CurrentUserService } from "src/app/shared/currentUser/current-user.service";
+import { DecisionDialogComponent } from "src/app/shared/dialogs/decision-dialog/decision-dialog.component";
+import { ContextMenuActionEvent, ContextMenuItem } from "src/app/shared/generic-table/generic-table.component";
+import { TpHeaderService } from "src/app/shared/tp-header/tp-header.service";
+
+/**
+ * RegionsTableComponent is the controller for the "Regions" table.
+ */
+@Component({
+	selector: "tp-regions",
+	styleUrls: ["./regions-table.component.scss"],
+	templateUrl: "./regions-table.component.html"
+})
+export class RegionsTableComponent implements OnInit {
+	/** List of regions */
+	public readonly regions: Promise<Array<ResponseRegion>>;
+
+	constructor(private readonly route: ActivatedRoute, private readonly headerSvc: TpHeaderService, private readonly router: Router,
+		private readonly api: CacheGroupService, private readonly dialog: MatDialog, public readonly auth: CurrentUserService) {
+		this.fuzzySubject = new BehaviorSubject<string>("");
+		this.regions = this.api.getRegions();
+	}
+
+	/** Initializes table data, loading it from Traffic Ops. */
+	public ngOnInit(): void {
+		this.route.queryParamMap.subscribe(
+			m => {
+				const search = m.get("search");
+				if (search) {
+					this.fuzzControl.setValue(decodeURIComponent(search));
+					this.updateURL();
+				}
+			},
+			e => {
+				console.error("Failed to get query parameters:", e);
+			}
+		);
+		this.headerSvc.headerTitle.next("Regions");
+	}
+
+	/** Definitions of the table's columns according to the ag-grid API */
+	public columnDefs = [
+		{
+			field: "name",
+			headerName: "Name"
+		},
+		{
+			field: "divisionName",
+			headerName: "Division"
+		},
+		{
+			field: "id",
+			headerName:" ID",
+			hide: true
+		},
+		{
+			field: "lastUpdated",
+			headerName: "Last Updated"
+		}
+	];
+
+	/** Definitions for the context menu items (which act on augmented region data). */
+	public contextMenuItems: Array<ContextMenuItem<Region>> = [
+		{
+			action: "edit",
+			multiRow: false,
+			name: "Edit"
+		},
+		{
+			action: "delete",
+			multiRow: false,
+			name: "Delete"
+		},
+		{
+			action: "viewRegions",
+			multiRow: false,
+			name: "View Regions"
+		}
+	];
+
+	/** A subject that child components can subscribe to for access to the fuzzy search query text */
+	public fuzzySubject: BehaviorSubject<string>;
+
+	/** Form controller for the user search input. */
+	public fuzzControl: FormControl = new FormControl<string>("");
```

Comment:
I _think_ this is abandoning any typing on the `FormControl` because it's not specifying the type parameter. That may be wrong, but in any case the type of a variable shouldn't be broader than the the type of the expression to which it's being assigned, in general. On the other hand, the expression is a simple constructor call, so the type is trivially inferred from the type of the expression, so you could also just leave the type annotation off entirely.

Reply:
If type sting is removed, it does complain with `TS1099: Type argument list cannot be empty.`

I don't know of another way to remove that inference to `FormControl`, especially if we want to simplify it's expression based on this description of `FormControl`:
```ɵFormControlCtor. <string>(     value: FormControlState<string> | string,     validatorOrOpts?: ValidatorFn | ValidatorFn[] | FormControlOptions | null,     asyncValidator?: AsyncValidatorFn | AsyncValidatorFn[] | null):    FormControl<string | null>
Various available constructors for FormControl. Do not use this interface directly. Instead, use FormControl:
const fc = new FormControl('foo');
This symbol is prefixed with ɵ to make plain that it is an internal symbol.
  @angular/forms```

Reply:
Well yeah, if you remove `FormControl` you also have to remove the <kbd>:</kbd> that tells TypeScript to expect a type expression.